src/main/java/gift/controller/admin/AdminProductController.java

```diff
@@ -0,0 +1,68 @@
+package gift.controller.admin;
+
+import gift.dto.ProductRequest;
+import gift.dto.ProductResponse;
+import gift.service.ProductService;
+import java.util.List;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Controller
+@RequestMapping("/admin/products")
+public class AdminProductController {
+
+    private final ProductService productService;
+
+    public AdminProductController(ProductService productService) {
+        this.productService = productService;
+    }
+
+    @GetMapping
+    public String showProductList(Model model) {
+        List<ProductResponse> products = productService.getAllProducts();
+        model.addAttribute("products", products);
+        return "admin/product/list";
+    }
+
+    @GetMapping("/new")
+    public String showCreateForm(Model model) {
+        model.addAttribute("product", new ProductRequest());
+        return "admin/product/form";
+    }
+
+    @PostMapping
+    public String createProduct(@ModelAttribute ProductRequest request) {
+        productService.addProduct(request);
+        return "redirect:/admin/products";
+    }
+
+    @GetMapping("/{id}/edit")
+    public String showEditForm(@PathVariable Long id, Model model) {
+        ProductResponse product = productService.findProductById(id);
+        ProductRequest request = new ProductRequest();
+        request.setId(product.getId());
+        request.setName(product.getName());
+        request.setPrice(product.getPrice());
+        request.setImageUrl(product.getImageUrl());
+
+        model.addAttribute("product", request);
```

Comment:
생각해보니 이 부분은 굳이 **ProductResponse** 를 `ProductRequest`로 변경하여 ModelAttribute로 넘겨줄 필요가 없을 것 같은데요? 🤔

조회해온 `ProductResponse` 를 그대로 product Attribute로 전달하면 되지 않을까요? 

```suggestion
        model.addAttribute("product", product);
```

Reply:
edit 페이지라서 수정된 데이터 내용을 클라이언트에서 서버로 전달해주는 것만 생각했었는데
생각해보니 서버에서 클라이언트로 edit 페이지 폼을 보내주는 역할이었네요!
제안해주신 대로 수정했습니다. 감사합니다!